ratis-server-api/src/main/java/org/apache/ratis/server/leader/LeaderState.java

```diff
@@ -66,4 +67,13 @@ AppendEntriesRequestProto newAppendEntriesRequestProto(FollowerInfo follower,
   /** Received an {@link AppendEntriesReplyProto} */
   void onAppendEntriesReply(LogAppender appender, AppendEntriesReplyProto reply);
 
+  /** Received a TransferLeadershipRequest */
+  void onTransferLeadership(RaftPeerId transferee);
+
+  /** Abort the last TransferLeadership */
+  void abortTransferLeadership();
+
+  /** Check if TransferLeadership is in progress */
+  boolean isTransferLeadershipInProgress();
+
```

Comment:
This are internal APIs -- all methods are used only inside the `org.apache.ratis.server.impl` package.  So, they should not be added here.